kge/util/dump.py

```diff
@@ -113,83 +113,167 @@ def _add_dump_trace_parser(subparsers_dump):
+            " add a respective column in the csv file with name 'new_key_name'."
+            " Additionally, for 'key_name' the special keys '$folder', '$machine'"
+            " '$checkpoint' and '$base_model' can be used. "
+        ),
+    )
+    parser_dump_trace.add_argument(
+        "--keys",
+        "-k",
+        nargs="*",
+        type=str,
+        help=(
+            "A list of 'key' entries (separated by space). For every entry,"
+            " the corresponding values of 'key' will be searched in the config and"
+            " trace entries and in the csv file a column with name 'key' will be added."
+            " The same special keys as for --keysfile can be used but must be"
+            " surrounded by quotations ''. "
+        ),
+    )
     parser_dump_trace.add_argument(
         "--checkpoint",
         default=False,
         action="store_const",
         const=True,
         help=(
-            "If source is a path to a job folder and --checkpoint is set the best "
-            "(if present) or last checkpoint will be used to determine the job_id"
+            "If source is a path to a job folder and --checkpoint is set, the best"
+            " (if present) or last checkpoint will be used to determine the job_id from"
+            " where the tracefile will be processed backwards."
         ),
     )
-
     parser_dump_trace.add_argument(
         "--job_id",
         default=False,
         help=(
-            "Specifies the training job id in the trace "
-            "from where to start processing backward"
+            "Specifies the training job id in the trace"
+            " from where to start processing the tracefile backward."
         ),
     )
-
     parser_dump_trace.add_argument(
         "--max_epoch",
         default=False,
         help=(
-            "Specifies the epoch in the trace"
-            "from where to start processing backwards"
+            "Specifies the max epoch number in the tracefile"
+            " from where to start processing backwards."
         ),
     )
-
     parser_dump_trace.add_argument(
         "--truncate",
         default=False,
         action="store_const",
         const=True,
         help=(
-            "If a checkpoint is used (by providing one explicitly as source or by "
-            "using --checkpoint), --truncate will define the max_epoch to process as"
-            "specified by the checkpoint"
+            "If a checkpoint is used (by providing one explicitly as source or by"
+            " using --checkpoint), --truncate will define the max_epoch number to"
+            " process as provided by the checkpoint."
         ),
     )
-
-    for argument in [
-        "--train",
-        "--valid",
-        "--test",
-        "--search",
+    parser_dump_trace.add_argument(
         "--yaml",
+        action="store_const",
+        const=True,
+        default=False,
+        help="Dump the full tracefile to stdout.",
```

Comment:
Not true. Just means do not use CSV, but full trace+config entries. Which entries depends on the other options.